pkg/apis/marketplace/v1alpha1/catalogsourceconfig_types.go

```diff
@@ -129,5 +134,19 @@ func (csc *CatalogSourceConfig) RemoveOwner(ownerUID types.UID) {
 
 // GetPackageIDs returns the list of package(s) specified.
 func (spec *CatalogSourceConfigSpec) GetPackageIDs() []string {
-	return strings.Split(strings.Replace(spec.Packages, " ", "", -1), ",")
+	return getValidPackageSliceFromString(spec.Packages)
+}
+
+func getValidPackageSliceFromString(pkgs string) []string {
+	pkgIds := make([]string, 0)
+
+	pkgSlice := strings.Split(strings.Replace(pkgs, " ", "", -1), ",")
```

Comment:
Is `strings.Replace` required If we sanitize each package name by trimming white spaces? For a leading comma `strings.Split` returns a package name with a space.

Reply:
If we do it to the initial string once that will remove any extra whitespace, meaning we don't need to call `strings.Replace()` on each token.

Reply:
Additionally I prefer this approach because removing all of the whitespace is needed anyway. This also avoids the issue of spaces in package names.